arcsi/api/item.py

```diff
@@ -177,24 +179,37 @@ def add_item():
 
         # TODO get show cover img and set as fallback
         if request.files:
+            # Defend against possible duplicate files
+            name_occurrence = int(db.session.query(db.func.count()).filter(new_item.name == Item.name, new_item.number == Item.number).scalar())
+            version_pre = None
+
+            if name_occurrence:
+                version_prefix = uuid4()
+                item_name = "{}-{}".format(version_prefix, new_item.name)
```

Comment:
@pvj how about putting the uuid4 in the end (as a postfix)? The advantage would be that if somebody downloads the file, the show and episode names are easily visible.

Reply:
Changed so that hash is appended as file name postfix: 8657aa46b803b54ae5152800e873a3960b44d081